webapp/src/components/Competency.tsx

```diff
@@ -1,15 +1,32 @@
-import React from 'react';
+import { Button } from '@mui/material';
+import React, { useContext, useState } from 'react';
+import CreateCompetencyForm from './CreateCompetencyForm';
+import SessionContext from './SessionContext';
+
+import { EntityId } from '../types/api';
 
 interface CompetencyProps {
-  label: string;
   description: string;
+  id: EntityId;
+  label: string;
+  onCompetenciesChanged?: (id: EntityId) => void;
+  principal_id: EntityId;
 }
 
-const Competency = ({ label, description }: CompetencyProps) => (
-  <div>
-    <h2>{label}</h2>
-    <p>{description}</p>
-  </div>
-);
+const Competency = ({ description, id, label, onCompetenciesChanged, principal_id }: CompetencyProps) => {
+  const { principalId } = useContext(SessionContext);
```

Comment:
`principalId` here would shadow the `principalId` prop. It might be worth aliasing this to `sessionPrincipalId` or something like that in order to differentiate it.

Reply:
Thanks, `sessionPrincipalId` would make a lot more sense.  To be sure, do you mean that this name would shadow the `principal_id` prop declared on line 13?  Thanks.